.ci/dockers/ubuntu-14.04/Dockerfile

```diff
@@ -0,0 +1,48 @@
+FROM microsoft/vsts-agent:ubuntu-14.04
```

Comment:
@guolinke Maybe move with file into `LightGBM/docker` folder?

Reply:
I think this docker is only for CI, not related to LightGBM's functions. Moving it to the main docker may cause confusing.

Reply:
Got it!